### PR TITLE
Reset checkout timer at refresh

### PR DESCRIPTION
**Pull request by cnohall**
Related to Issue #251.

The code has seen small changes, but it should be enough to implement the proposed changes. Every time a user refreshes the page, an API call will be sent to check and update the price, and the timer will be reset.

### CODE REVIEW

**Review by DarrenWestwood — COMMENTED** [review]
Working as desired.

The time_remaining column should be removed for upgrading users. Architecture for database upgrades can be found here: https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L511-L514

File: php/Blockonomics.php [hunk 1]
```diff
@@ -466,9 +466,8 @@ public function calculate_order_params($order){
                 $price = 1;
             }
             $order['satoshi'] = intval(round(1.0e8*$wc_order->get_total()/$price));
-            $order['timestamp'] = time();
```
[the remark this answers is not in the document]
Comment (cnohall) [reply php/Blockonomics.php:469]: It has now been reinstated

File: blockonomics-woocommerce.php [hunk 2]
```diff
@@ -531,10 +530,10 @@ function blockonomics_update_db_check() {
 
     $installed_ver = get_site_option( 'blockonomics_db_version' );
     if ( $installed_ver != $blockonomics_db_version ) {
-        // Example function to demonstrate table changes between upgrade versions
-        // if ($installed_ver < 1.0) {
-        //     $wpdb->query("ALTER TABLE $table_name DROP transaction;");
-        // }
+        $table_name = $wpdb->prefix . 'blockonomics_orders';
+        if ($blockonomics_db_version > 1.0) {
```
Comment (DarrenWestwood) [inline blockonomics-woocommerce.php:534]: This code will run again incorrectly when `if ($blockonomics_db_version > 1.1)` is added and the user upgrades from db_version 1.1->1.2.
We should use `if ($installed_ver < 1.1)` as we want to run the upgrades between the installed version and the new db_version. This also makes the code easier to read as the if statement references the matching db_version

Comment (cnohall) [reply blockonomics-woocommerce.php]: Ah, that makes more sense. Thanks for the clarification! I've pushed the change

**Review by DarrenWestwood — APPROVED** [review]

File: blockonomics-woocommerce.php [hunk 3]
```diff
@@ -531,10 +530,10 @@ function blockonomics_update_db_check() {
 
     $installed_ver = get_site_option( 'blockonomics_db_version' );
     if ( $installed_ver != $blockonomics_db_version ) {
-        // Example function to demonstrate table changes between upgrade versions
-        // if ($installed_ver < 1.0) {
-        //     $wpdb->query("ALTER TABLE $table_name DROP transaction;");
-        // }
+        $table_name = $wpdb->prefix . 'blockonomics_orders';
+        if ($installed_ver < 1.1) {
+            $wpdb->query("ALTER TABLE $table_name DROP time_remaining;");
```
Comment (blockonomics) [inline blockonomics-woocommerce.php:535]: As far I remember dbDelta will automatically drop the column when `blockonomics_create_table` is called so this is uncessary. Please verify with real table having values

File: blockonomics-woocommerce.php [hunk 4]
```diff
@@ -501,7 +501,6 @@ function blockonomics_create_table() {
         crypto varchar(3) NOT NULL,
         address varchar(191) NOT NULL,
         timestamp int,
```
Comment (blockonomics) [inline blockonomics-woocommerce.php:503]: What is the use of `timestamp` column now?  If its not being used anymore better to remove this also

**Review by blockonomics — CHANGES_REQUESTED** [review]

File: blockonomics-woocommerce.php (hunk 3 above)
Comment (cnohall) [reply blockonomics-woocommerce.php:535]: It seems like dbDelta cannot delete columns. ([source](https://wordpress.stackexchange.com/a/76930))

File: blockonomics-woocommerce.php (hunk 4 above)
Comment (cnohall) [reply blockonomics-woocommerce.php]: true, now the `timestamp` has been dropped

**Comment by cnohall** [comment]
The requested changes have been done

**Pull request merged**